src/app/report/edit-display/edit-display.component.ts

```diff
@@ -85,22 +90,33 @@ export class EditDisplayComponent {
     private errorHandler: ErrorHandling,
   ) {}
 
-  showReport(report: Report): void {
+  showReport(node: ReportOrCheckpoint): void {
     this.disableEditing();
-    this.report = report;
-    report.xml
-      ? this.editor.setNewReport(report.xml)
-      : this.editor.setNewReport(this.helperService.convertMessage(report));
+    this.selectedNode = node;
+    if (ReportUtil.isReport(this.selectedNode)) {
+      this.editor.setNewReport(this.selectedNode.xml);
+    } else if (ReportUtil.isCheckPoint(this.selectedNode)) {
+      this.editor.setNewReport(this.convertMessage(this.selectedNode));
+    }
     this.rerunResult = undefined;
     this.displayReport = true;
   }
 
-  changeEncoding(button: any): void {
-    this.editor.setNewReport(this.helperService.changeEncoding(this.report, button));
+  convertMessage(checkpoint: Checkpoint): string {
+    let message: string = checkpoint.message === null ? '' : checkpoint.message;
```

Comment:
```suggestion
    let message: string = checkpoint.message ?? '';
```
I think this is a cleaner way to write this line of code, right?

Reply:
Yes, I had not noticed that line yet. I added the change.